### PR TITLE
Index location fixture

### PR DESCRIPTION
**Pull request by esoergel**
This adds a schema block for the locations fixture, telling mobile to index the appropriate attributes.  This is based on https://github.com/dimagi/commcare-core/pull/502
@proteusvacuum @czue @ctsims 

@ctsims how confident are you that this won't break anything?  I'll throw this up on staging and see about loading an app or two.

### CODE REVIEW

File: corehq/apps/locations/tests/data/index_location_fixtures.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<partial>
```
Comment (esoergel) [inline corehq/apps/locations/tests/data/index_location_fixtures.xml]: Here's what the output format looks like.

File: corehq/apps/locations/tests/test_location_fixtures.py [hunk 2]
```diff
@@ -34,24 +34,27 @@ class FixtureHasLocationsMixin(TestXmlMixin):
...
+
+    def _assert_fixture_has_locations(self, xml_name, desired_locations, flat=False):
+        generator = flat_location_fixture_generator if flat else location_fixture_generator
+        fixture = ElementTree.tostring(generator(self.user, V2)[-1])
+        desired_fixture = self._assemble_expected_fixture(xml_name, desired_locations)
         self.assertXmlEqual(desired_fixture, fixture)
 
 
-@mock.patch.object(Domain, 'uses_locations', return_value=True)  # removes dependency on accounting
+@mock.patch.object(Domain, 'uses_locations', lambda: True)  # removes dependency on accounting
```
Comment (proteusvacuum) [inline corehq/apps/locations/tests/test_location_fixtures.py]: 👍

File: corehq/apps/locations/fixtures.py [hunk 3]
```diff
@@ -124,7 +128,17 @@ def get_xml_nodes(self, fixture_id, restore_user, all_locations):
             _fill_in_location_element(location_node, location)
             outer_node.append(location_node)
 
-        return [root_node]
+        return root_node
+
+    def _get_schema_node(self, fixture_id, location_type_attrs):
+        indices_node = Element('indices')
+        for index_attr in location_type_attrs + ['id', 'type']:
```
Comment (proteusvacuum) [inline corehq/apps/locations/fixtures.py]: nit: could pull these out into a named constant

**Review by proteusvacuum — APPROVED** [review]

**Review by czue — APPROVED** [review]
nice thanks for knocking this out so quickly! 

build fail looks real.

```

======================================================================
FAIL: test_index_location_fixtures (corehq.apps.locations.tests.test_location_fixtures.LocationFixturesTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/locations/tests/test_location_fixtures.py", line 253, in test_index_location_fixtures
    self.assertXmlEqual(expected_schema, schema)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -1,9 +1,9 @@
 <partial>
   <schema id="locations">
     <indices>
+      <index>@state_id</index>
+      <index>@county_id</index>
       <index>@city_id</index>
-      <index>@county_id</index>
-      <index>@state_id</index>
       <index>@id</index>
       <index>@type</index>
     </indices>
```

**Comment by esoergel** [comment]
ugh, I'll fix that build error - the order is non-deterministic, so I'll have to sort or something.

Meanwhile, I deployed this to staging and deployed to an app without issue.
[app](https://staging.commcarehq.org/a/ethan-test/apps/view/ab81ce0c2aaf8df0775ff6712722c72a/releases/)
[restore](https://staging.commcarehq.org/hq/admin/phone/restore/?version=2.0&as=ethan@ethan-test.commcarehq.org)

File: corehq/apps/locations/fixtures.py (hunk 3 above)
Comment (esoergel) [inline corehq/apps/locations/fixtures.py]: :+1:

**Comment by czue** [comment]
awesome! would be great if we can get this deployed to india today

**Comment by esoergel** [comment]
I can do that.

**Comment by proteusvacuum** [comment]
📈 🐎

**Pull request merged**